### PR TITLE
Use drmGetDevices to list nodes

### PR DESCRIPTION
**Pull request by emersion**
There's no guarantee that DRM primary nodes start at card0. I've run into a
system where the first card is card1 (probably because the driver has been
unloaded/reloaded?). Because of this drm_info wasn't dumping any data.

Instead, we can use drmGetDevices which returns a list of DRM devices and nodes
available on the system.

### CODE REVIEW

File: json.c [hunk 1]
```diff
@@ -628,18 +628,28 @@ struct json_object *drm_info(char *paths[])
...
+			perror("drmGetDevices");
+			json_object_put(obj);
+			return NULL;
+		}
 
-			struct json_object *dev = node_info(path);
-			if (!dev)
+		for (int i = 0; i < n; ++i) {
+			drmDevice *dev = devices[i];
+			if (!(dev->available_nodes & (1 << DRM_NODE_PRIMARY)))
```
Comment (emersion) [inline json.c]: See https://gitlab.freedesktop.org/mesa/drm/blame/master/xf86drm.c#L3364

File: json.c [hunk 2]
```diff
@@ -628,18 +628,28 @@ struct json_object *drm_info(char *paths[])
...
 				continue;
 
-			json_object_object_add(obj, path, dev);
+			const char *path = dev->nodes[DRM_NODE_PRIMARY];
```
Comment (emersion) [inline json.c]: See https://gitlab.freedesktop.org/mesa/drm/blame/master/xf86drm.c#L3376

File: json.c [hunk 3]
```diff
@@ -628,18 +628,28 @@ struct json_object *drm_info(char *paths[])
 
 	/* Print everything by default */
 	if (!paths[0]) {
-		char path[PATH_MAX];
-		for (int i = 0;; ++i) {
-			snprintf(path, sizeof path, DRM_DEV_NAME, DRM_DIR_NAME, i);
-			if (access(path, R_OK) < 0)
-				break;
+		drmDevice *devices[64];
```
Comment (emersion) [inline json.c]: Ah, and see https://gitlab.freedesktop.org/mesa/drm/blame/master/xf86drm.c#L3799 for this one

**Comment by ascent12** [comment]
Thanks

**Pull request merged**